docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/efficientdet_pipeline.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2021 Kacper Kluk, Paweł Anikiel, Jagoda Kamińska. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import nvidia.dali as dali
+import nvidia.dali.plugin.tf as dali_tf
+import tensorflow as tf
+import math
+
+from absl import logging
+from glob import glob
+from pipeline import anchors
+from utils import InputType
+
+from . import ops_util
+
+
+class EfficientDetPipeline:
+    def __init__(
+        self,
+        params,
+        batch_size,
+        args,
+        is_training=True,
+        num_shards=1,
+        device_id=0,
+        cpu_only=False,
+    ):
+        self._batch_size = batch_size
+        self._image_size = params["image_size"]
+        self._gridmask = params["grid_mask"]
+
+        self._input_type = args.input_type
+        if self._input_type == InputType.tfrecord:
+            if is_training:
+                file_pattern = args.train_file_pattern
+            else:
+                file_pattern = args.eval_file_pattern or args.train_file_pattern
+
+            self._tfrecord_files = glob(file_pattern)
+            self._tfrecord_idxs = [
+                filename + "_idx" for filename in self._tfrecord_files
+            ]
+        elif self._input_type == InputType.coco:
+            self._images_path = args.images_path
+            self._annotations_path = args.annotations_path
+
+        self._is_training = is_training
+        self._num_shards = num_shards
+        self._shard_id = None if cpu_only else device_id
+        self._device = "cpu" if cpu_only else "gpu"
+
+        self._anchors = anchors.Anchors(
+            3, 7, 3, [1.0, 2.0, 0.5], 4.0, params["image_size"]
+        )
+        self._boxes = self._get_boxes()
+        self._max_instances_per_image = params["max_instances_per_image"] or 100
+        seed = params["seed"] or -1
+
+        self._pipe = dali.pipeline.Pipeline(
+            batch_size=self._batch_size,
+            num_threads=self._num_shards,
+            device_id=device_id,
+            seed=seed,
+        )
+        self._define_pipeline()
```

Comment:
Can we apply our decorator to members? If so, then how about decorating _define_pipeline with `@pipeline_def` and having
```suggestion
        self._pipe = self._define_pipeline(
            batch_size=self._batch_size,
            num_threads=self._num_shards,
            device_id=device_id,
            seed=seed,
        )
```

Reply:
Done